pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2404,11 +2404,19 @@ func buildListener(port api.ServicePort, annotations map[string]string, sslPorts
 	return listener, nil
 }
 
+func nodeNames(nodes []*api.Node) sets.String {
```

Comment:
(This one is actually different (`sets.String` vs `[]string`) but yes there are several duplicates of this function)

I thought about that as I was typing it, but decided against sharing a common version because it's trivial, shouldn't really be part of a public/standardised API, and I expect it will disappear once the provider authors properly refactor around the new `Node` datatype.  I'm happy to spin it out into a new provider utils.go or similar if you think it's important enough.